scripts/setup_android_env.sh

```
@@ -40,4 +45,72 @@ if [ -n "${CLAUDE_PROJECT_DIR:-}" ] && [ ! -f "$CLAUDE_PROJECT_DIR/local.propert
+
+  if ! curl -fsSL -o "$dist_dir/$zip_name" "$url"; then
+    echo "WARN: could not download $url" >&2
+    echo "      The network policy must allow the services.gradle.org ->" >&2
+    echo "      github.com release-asset redirect during env setup." >&2
+    rm -f "$dist_dir/$zip_name"
+    return 0
+  fi
+
+  unzip -q -o "$dist_dir/$zip_name" -d "$dist_dir"
+  touch "$marker"
+  echo "Gradle wrapper ready: $(ls -d "$dist_dir"/*/ 2>/dev/null | head -1)"
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)